src/shared/tpm2-util.c

```diff
@@ -311,55 +321,222 @@ static int tpm2_credit_random(ESYS_CONTEXT *c) {
+                                .parameters.rsaDetail = {
+                                        .symmetric = {
+                                                .algorithm = TPM2_ALG_AES,
+                                                .keyBits.aes = 128,
+                                                .mode.aes = TPM2_ALG_CFB,
+                                        },
+                                        .scheme.scheme = TPM2_ALG_NULL,
+                                        .keyBits = 2048,
+                                },
+                        },
+                },
+                /* Index 3 ECC SRK */
+                {
+                        .publicArea = {
+                                .type = TPM2_ALG_ECC,
+                                .nameAlg = TPM2_ALG_SHA256,
+                                .objectAttributes = TPMA_OBJECT_FIXEDTPM|TPMA_OBJECT_FIXEDPARENT|TPMA_OBJECT_SENSITIVEDATAORIGIN|TPMA_OBJECT_RESTRICTED|TPMA_OBJECT_DECRYPT|TPMA_OBJECT_USERWITHAUTH|TPMA_OBJECT_NODA,
+                                .parameters.eccDetail = {
+                                        .symmetric = {
+                                                .algorithm = TPM2_ALG_AES,
+                                                .keyBits.aes = 128,
+                                                .mode.aes = TPM2_ALG_CFB,
+                                        },
+                                        .scheme.scheme = TPM2_ALG_NULL,
+                                        .curveID = TPM2_ECC_NIST_P256,
+                                        .kdf.scheme = TPM2_ALG_NULL,
                                 },
-                                .scheme.scheme = TPM2_ALG_NULL,
-                                .keyBits = 2048,
                         },
                 },
         };
 
+        /* make sure if someone uses something other than true false we don't explode */
+        return &templ[((use_srk_model & 0x1) << 1) + (is_ecc & 0x1)];
+}
+
+#define SRK_HANDLE 0x81000001
+/*
+ * See tpm2-tss bug: https://github.com/tpm2-software/tpm2-tss/issues/2553
+ * Keep it simple, just use the raw numerics.
+ */
+#define TPM2_HR_PERSISTENT_FIRST_GOOD  0x81000000
+
+static int tpm2_get_srk(
+                ESYS_CONTEXT *c,
+                ESYS_TR *ret_primary,
+                TPMI_ALG_PUBLIC alg,
+                TPMI_ALG_PUBLIC *ret_alg) {
+
+        TPMI_YES_NO more_data;
+        ESYS_TR primary = ESYS_TR_NONE;
+        _cleanup_(Esys_Freep) TPMS_CAPABILITY_DATA *cap_data = NULL;
+
+        TSS2_RC rc = sym_Esys_GetCapability(c,
+                        ESYS_TR_NONE,
+                        ESYS_TR_NONE,
+                        ESYS_TR_NONE,
+                        TPM2_CAP_HANDLES,
+                        TPM2_HR_PERSISTENT_FIRST_GOOD,
+                        TPM2_MAX_CAP_HANDLES,
+                        &more_data,
+                        &cap_data);
```

Comment:
This is stupid, I should just ask the tpm for a list of handles of Len 1 starting at SRK. If the retuned list is empty, not there, else present... duh.